src/Orleans/Threading/GrainCancellationToken.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Threading;
+using Orleans.CodeGeneration;
+using Orleans.Runtime;
+using Orleans.Serialization;
+
+namespace Orleans.Threading
```

Comment:
No need for a new Treading folder and new namespace. Namespaces were designed with lots of thought. We dont just create random **public** namespaces (internal impl. is different).
Put it under Async folder, Orleans namespace. It is a new public async type.